### DISCUSSION TITLE
Incorrect URL highlighting

### DISCUSSION THREAD

**Discussion by TimReset** [gsantner/markor#2089] [Q&A] [answered]
Hi!
I noticed Markor highlights some paths as URL:
eg /test/test.test/
test.test/ - will be showed as URL. That is incorrect.
![image](https://github.com/gsantner/markor/assets/911949/5c25a8c9-a69f-4811-af43-12fd19e41674)
I see Markor uses android.util.Patterns#WEB_URL and this is doubly strange for me - I thought Android bundled pattern should work correctly. 
So, can we use more strict patter to find URL? (eg with required protocol - http(s), ftp, etc)
I can create pr but just to want to be sure it's an inconvenient not only for me. I'm working in Termux and it uses /data/data/com.termux/ paths - Markor makes all of them as URL in my notes :-(

**Comment by gsantner** [comment]
For syntax highlighting links a rather short regex is used for performance reason.
For rendering "view mode" of course   something better is in place. Highlighting should be instant while the other can take the time it needs.

As you wrote about ftp, http and WEB_URL I guess you have an rough idea how many variations and rules there are.

And stuffing every rule into regex comes at expense of performance. So the current approach is rather a relaxed detection with a low performance penalty.


I rather have something that fast highlights 90% fine and 10% wrong then 99% fine and slow.
If you can find something better not too complex it would be fine.

Hope it answers all questions :).

**Reply by gsantner** [reply]
Oh, I see that really Android builtin WEB_URL is used by now for that. I recall a previous implementation where a custom crafted regex was in place 😄 .

For the WEB_URL seems like it is especially better suited towards validating a link or checking a single link in a text, rather then whole file.

**Reply by TimReset** [reply]
So, is it reasonable to use a custom pattern to detect URLs inside text?

**Reply by gsantner** [reply, marked as answer]
@TimReset Seems like a different pattern might suit better. But as you can see, even a very clever pattern where many people put their mind on it...may fail 😄 .

**Reply by TimReset** [reply]
I wrote and use a custom patter now - looks fine. I'll create PR with examples.

**Comment by Asmageddon** [comment]
The URL highlighting in edit mode is really obnoxious, since it changes text dimensions and makes it harder to read. I'd really just rather not have it.

**Reply by gsantner** [reply]
It is actually kind of wanted - so 4 textline long links draw less attention and you can look at the rest of the real text. If you want to edit URLs, open them, click on them...you have both the option to use the Link editor as well to use the view mode.

**Reply by Asmageddon** [reply]
It's extremely, extremely unwanted when you have random "word.word" stuff being compressed down and it's all flickering during editing and way harder to select and copy while the actual length savings have negligible positive impact. It's also distracting when I have "domain.com" inside a paragraph or header, and it's styled differently.

If it's intended to compress long links, why not apply it only to particularly long ones? Or at least match URLs with a regex better than two words having a dot inbetween.
